contrib/linux-kernel/lib/zstd/compress.c

```diff
@@ -587,8 +587,10 @@ ZSTD_STATIC size_t ZSTD_compressSequences(ZSTD_CCtx *zc, void *dst, size_t dstCa
 {
 	const int longOffsets = zc->params.cParams.windowLog > STREAM_ACCUMULATOR_MIN;
 	const seqStore_t *seqStorePtr = &(zc->seqStore);
-	U32 count[MaxSeq + 1];
-	S16 norm[MaxSeq + 1];
+	U32 *count = (U32 *)zc->tmpCounters;
+	S16 *norm = (S16 *)(count + MaxSeq + 1);
+	U32 *const workspace = PTR_ALIGN((U32 *)(norm + MaxSeq + 1), sizeof(U32));
```

Comment:
I know this is supposed to work, since `PTR_ALIGN` makes a good job at properly aligning the resulting ptr,
but still, from a compiler perspective, promoting a `U16*` into a `U32*` is ground for a `-Wcast-align` warning :
https://godbolt.org/g/vy7fR3